x-pack/filebeat/module/crowdstrike/falcon/config/pipeline.js

```diff
@@ -0,0 +1,132 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+var crowdstrikeFalcon = (function() {
+    var processor = require("processor");
+
+    var convertUnderscore = function(text) {
+        return text.split(/(?=[A-Z])/).join('_').toLowerCase(); 
+    };
+
+    var decodeJson = new processor.DecodeJSONFields({
+        fields: ["message"],
+        target: "crowdstrike",
+        process_array: true,
+        max_depth: 8
+    });
+
+    var dropFields = function(evt) {
+        evt.Delete("message");
+        evt.Delete("host.name");
+    };
+  
+    var setFields = function (evt) {
+        evt.Put("agent.name", "falcon");
+    };
+
+    var convertFields = new processor.Convert({
+        fields: [
+            // DetectionSummaryEvent
+            { from: "crowdstrike.event.LocalIP", to: "source.ip" },
+            { from: "crowdstrike.event.ProcessId", to: "process.pid" },
+
+            // UserActivityAuditEvent and AuthActivityAuditEvent
+            { from: "crowdstrike.event.UserIp", to: "source.ip" },
+        ],
+        mode: "copy",
+        ignore_missing: true,
+    });
+
+    var parseTimestamp = new processor.Timestamp({
+        field: "crowdstrike.metadata.eventCreationTime",
+        target_field: "@timestamp",
+        timezone: "UTC",
+        layouts: ["UNIX_MS"],
+        ignore_missing: false,
+    });
+
+    var processEvent = function(evt) {
+        var eventType = evt.Get("crowdstrike.metadata.eventType")
+
+        evt.Put("agent.type", "falcon")
+        evt.Put("event.kind", "event")
+        evt.Put("host.name", "")
```

Comment:
This is a good question. I was unable to delete the field. It's added later by filebeat. So it's misleading in the UI to see an event that came from some other endpoint associated with the filebeat agent. I thought it would be better to show an empty field in the UI to not mislead a user. But I'm 100% open to suggestions.

Reply:
Oh I thought I had amended that comment. I realized after the fact that in your description you say this log would be collected centrally, so disregard my comment on "running Filebeat on the host".

But we'll have to figure out what to do, to fill `host.name` though.

Filling it with the server's name for their server audit logs is the right thing to do as this is where these events are being reported. Then overwriting it with the event's computername for endpoint events is the right thing to do as well.

Reply:
> Broader discussion point: as I note in one of my comments, this seems to be a log that mixes their service's audit logs with actual endpoint detection events. Both are worth capturing, of course. But I wonder if we should find a way to separate them out in two different `event.dataset`, as their meaning & semantics are pretty different.

This is a good suggestion. Curious for suggestions, here.

Maybe ```crowdstrike.falcon.events``` and ```crowdstrike.falcon.auditlogs```?

Reply:
My understanding is that it's usually tied to each subdirectory of a given module (e.g. Apache HTTPD has two sister directories under https://github.com/elastic/beats/tree/master/filebeat/module/apache, one for "apache.access" logs, one for "apache.error" logs). So not sure if it's going to be enough to  simply overwrite `event.dataset` with two different values from a single pipeline, but it's worth a try.

Now for the values in `event.dataset`, ECS doesn't mandate anything. But the Filebeat convention is usually two levels. Perhaps something like "crowdstrike.endpoint" and "crowdstrike.server"? Not sure how important "Falcon" is in the branding, or if they have many other products or what. Perhaps "crowdstrike.falcon.endpoint" and "crowdstrike.falcon.server" would be fine as well. I'm not aware of anything that has a hard dependency on this convention.

Auditbeat for example fills `event.dataset` with a single word, and that hasn't been a problem, afaik.

Reply:
> This is a good question. I was unable to delete the field. It's added later by filebeat. So it's misleading in the UI to see an event that came from some other endpoint associated with the filebeat agent. I thought it would be better to show an empty field in the UI to not mislead a user. But I'm 100% open to suggestions.

Just an FYI, if you had done the Delete in an Elasticsearch pipeline it would have worked, that happens in ES after all the Filebeat processing.

Reply:
Yep. I tried to intentionally do nothing in ES to keep things simple for any users that might be using Logstash.